dask_kubernetes/objects.py

```diff
@@ -229,4 +229,56 @@ def clean_pod_template(pod_template):
     else:
         pod_template.spec.tolerations.extend(tolerations)
 
+    # add default node affinity to k8s.dask.org/node-purpose=worker
+    if match_node_purpose is not "ignore":
+        # for readability
+        affinity = pod_template.spec.affinity
+
+        if affinity is None:
+            affinity = client.V1Affinity()
+        if affinity.node_affinity is None:
+            affinity.node_affinity = client.V1NodeAffinity()
+
+        # a common object for both a preferred and a required node affinity
+        node_selector_term = client.V1NodeSelectorTerm(
+            match_expressions=[
+                client.V1NodeSelectorRequirement(
+                    key="k8s.dask.org/node-purpose", operator="In", values=["worker"]
```

Comment:
are we okay with In vs. Equals here?

Reply:
Logically right now they are exactly the same, I think it is possible to make it `"Equal"` (or was it `"Equals"`?) instead assuming we remove the brackets `[]` around `"worker"`.

Reply:
Looking at this a little more, it doesn't seem like Equals is allowed here: https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1NodeSelectorRequirement.md

Reply:
Ah